chain/client/src/sync/external.rs

```diff
@@ -7,6 +7,20 @@ use std::sync::Arc;
 use std::time::Duration;
 use std::time::Instant;
 
+pub enum ObjectType {
+    StatePart,
+    StateHeader,
+}
```

Comment:
Looking at usecases, I would consider giving StatePart a parameter `StatePart(part_id)`. This would help to deduplicate logic of `external_*_storage_location` by passing `ObjectType` to them and determine path based on that.

Also word "object" looks too abstract when we have context of State. What about `StateFileType` with types `Part` and `Header`?